### PR TITLE
feat: add `wasm_bindgen` interface

### PR DESCRIPTION
**Pull request by tbrezot**
Change the `Policy` format:
- keep backward compatibility;
- include `EncryptionHint`s for each attribute to specify which kind of encryption to use for data tagged with this attribute;
- add format version number

### CODE REVIEW

File: src/policy.rs [hunk 1]
```diff
@@ -80,10 +80,31 @@ impl PolicyAxis {
...
+    /// Policy axes: maps axes name to the list of associated attribute names
+    /// and a boolean defining whether or not this axis is hierarchical.
+    pub axes: HashMap<String, (Vec<String>, bool)>,
+    /// Maps an attribute to its values and its hybridization hint.
+    pub attributes: HashMap<Attribute, Vec<u32>>,
+}
+
+#[derive(Clone, Serialize, Deserialize, Debug, PartialEq, Eq, PartialOrd, Ord)]
+pub enum PolicyVersion {
+    V1,
```
Comment (ThibaudDauce) [inline src/policy.rs:120]: Should be V2, no?

Comment (tbrezot) [reply src/policy.rs:120]: The previous version is called `LegacyPolicy` and has no version number. If I use `V2` for the current version, there would be no `V1`.

**Review by ThibaudDauce — APPROVED** [review]

File: src/error.rs [hunk 2]
```diff
@@ -31,4 +33,13 @@ pub enum Error {
     InvalidAttribute(String),
     #[error("invalid axis: {0}")]
     InvalidAxis(String),
+    #[error("deserialization error: {0}")]
+    DeserializationError(String),
```
Comment (ThibaudDauce) [inline src/error.rs]: Should be a `SerdeError` instead of a `String`?

**Comment by ThibaudDauce** [comment]
Could you add a `impl From<Error> for JsValue` to avoid all these `map_err`?

**Comment by ThibaudDauce** [comment]
Is there a way to remove all these `.map_err(|e| JsValue::from_str(&e.to_string()))`? If there was some Rust function that return `Result<T, Error>` then you can use the `impl From<JsValue> for Error` to replace the `.map_err()` by `?`, no? Not sure if it's worth it…

File: src/policy.rs [hunk 3]
```diff
@@ -107,11 +128,38 @@ impl Display for Policy {
...
+                        axes: policy.axes,
+                        attributes: policy
+                            .attributes
+                            .into_iter()
+                            .map(|(name, values)| (name, (values, EncryptionHint::Classic)))
+                            .collect(),
+                    })
+                } else {
+                    Err(Error::DeserializationError(e))
+                }
```
Comment (ThibaudDauce) [inline src/policy.rs]: I think the `if let` is useless, no? Just use `?`?

Comment (tbrezot) [reply src/policy.rs]: That's true, now that I have added a dedicated error variant, thanks :+1:

Comment (tbrezot) [reply src/policy.rs]: Actually it is still useful because we are not returning the error from the `LegacyPolicy` deserialization but the `Policy` one.

Comment (ThibaudDauce) [reply src/policy.rs]: Not sure to understand what is the problem…

Comment (ThibaudDauce) [reply src/policy.rs]: Hum ok! Undertood :-) Maybe add a comment or rename `e` to something more specific?

Comment (tbrezot) [reply src/policy.rs]: Yes you are right, even I got mistaken be reading my own code.

**Comment by tbrezot** [comment]
> Is there a way to remove all these `.map_err(|e| JsValue::from_str(&e.to_string()))`? If there was some Rust function that return `Result<T, Error>` then you can use the `impl From<JsValue> for Error` to replace the `.map_err()` by `?`, no? Not sure if it's worth it…

I improved the error management. Implementing `From<JsValue> for Error` was not needed since we mostly convert `Error` into `JsValue`.

File: src/error.rs (hunk 2 above)
Comment (tbrezot) [reply src/error.rs]: Yes

**Review by Manuthor — APPROVED** [review]
LGTM

**Pull request merged**